server/src/main/java/org/opensearch/index/mapper/ObjectMapper.java

```diff
@@ -420,9 +417,13 @@ protected static void parseProperties(ObjectMapper.Builder objBuilder, Map<Strin
         } else {
             this.mappers = CopyOnWriteHashMap.copyOf(mappers);
         }
-        this.nestedTypePathAsString = "__" + fullPath;
-        this.nestedTypePathAsBytes = new BytesRef(nestedTypePathAsString);
-        this.nestedTypeFilter = new TermQuery(new Term(TypeFieldMapper.NAME, nestedTypePathAsBytes));
+        Version version = Version.indexCreated(settings);
```

Comment:
This is a bit risky call: when fixing one of the `Joda` conversions, realized that `settings` may not hold the index created version all the time [1], may be worth checking here as well.

[1] https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/mapper/Mapper.java#L258

Reply:
Did you find this to be true in production or could it have just been a test that didn't set the Version? I don't think we can have a case where an index is created without a version, otherwise rolling upgrades would fail.

Reply:
It is this one https://github.com/opensearch-project/OpenSearch/pull/2094/files

Reply:
Just getting back to this. The [testDefaultConfig](https://github.com/opensearch-project/OpenSearch/pull/3004/files#diff-7b7553a5370e5b27cb2427f71bd8a727467b91fe30fca0ef321e440c0596f577R25) I had in there would've failed since it uses prefabricated empty settings but it was updating w/ different mappings so I added an [explicit test](https://github.com/opensearch-project/OpenSearch/pull/3004/files#diff-7b7553a5370e5b27cb2427f71bd8a727467b91fe30fca0ef321e440c0596f577R37) based on #1921 to update with the same mappings and Empty settings. There are no issues; we may revisit if this rears it's head again.